plugins/wasm-rust/extensions/sse-timing/src/lib.rs

```diff
@@ -0,0 +1,195 @@
+use higress_wasm_rust::log::Log;
+use higress_wasm_rust::rule_matcher;
+use higress_wasm_rust::rule_matcher::{on_configure, RuleMatcher, SharedRuleMatcher};
+use proxy_wasm::traits::{Context, HttpContext, RootContext};
+use proxy_wasm::types::{ContextType, DataAction, HeaderAction, LogLevel};
+use serde::Deserialize;
+use std::cell::RefCell;
+use std::ops::DerefMut;
+use std::rc::Rc;
+use std::str::from_utf8;
+use std::time::{Duration, SystemTime};
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Trace);
+    proxy_wasm::set_root_context(|_|Box::new(SseTimingRoot::new()));
+}}
+
+struct SseTimingRoot {
+    log: Rc<Log>,
+    rule_matcher: SharedRuleMatcher<SseTimingConfig>,
+}
+
+struct SseTiming {
+    log: Rc<Log>,
+    rule_matcher: SharedRuleMatcher<SseTimingConfig>,
+    vendor: String,
+    is_event_stream: bool,
+    event_stream: EventStream,
+    start_time: SystemTime,
+}
+
+#[derive(Default, Clone, Debug, Deserialize)]
+struct SseTimingConfig {
+    vendor: Option<String>,
+}
+
+impl SseTimingRoot {
+    fn new() -> Self {
+        SseTimingRoot {
+            log: Rc::new(Log::new("sse_timing".to_string())),
+            rule_matcher: rule_matcher::new_shared(),
+        }
+    }
+}
+
+impl Context for SseTimingRoot {}
+
+impl RootContext for SseTimingRoot {
+    fn on_configure(&mut self, _plugin_configuration_size: usize) -> bool {
+        on_configure(
+            self,
+            _plugin_configuration_size,
+            self.rule_matcher.borrow_mut().deref_mut(),
+            &self.log,
+        )
+    }
+
+    fn create_http_context(&self, _context_id: u32) -> Option<Box<dyn HttpContext>> {
+        Some(Box::new(SseTiming {
+            log: self.log.clone(),
+            rule_matcher: self.rule_matcher.clone(),
+            vendor: "higress".into(),
+            is_event_stream: false,
+            event_stream: EventStream::new(),
+            start_time: self.get_current_time(),
+        }))
+    }
+
+    fn get_type(&self) -> Option<ContextType> {
+        Some(ContextType::HttpContext)
+    }
+}
+
+impl Context for SseTiming {}
+
+impl HttpContext for SseTiming {
+    fn on_http_request_headers(
+        &mut self,
+        _num_headers: usize,
+        _end_of_stream: bool,
+    ) -> HeaderAction {
+        self.start_time = self.get_current_time();
+
+        let binding = self.rule_matcher.borrow();
+        let config = match binding.get_match_config() {
+            None => {
+                return HeaderAction::Continue;
+            }
+            Some(config) => config.1,
+        };
+        match config.vendor.clone() {
+            None => {}
+            Some(vendor) => self.vendor = vendor,
+        }
+        HeaderAction::Continue
+    }
+
+    fn on_http_response_headers(
+        &mut self,
+        _num_headers: usize,
+        _end_of_stream: bool,
+    ) -> HeaderAction {
+        match self.get_http_response_header("Content-Type") {
```

Comment:
The E2E test failed because it modifies the body; hence, it is necessary to remove the 'Content-Length' header during the header processing stage.

Reply:
According to text/event-stream MIME type, the origin server should not set Content-Length in response headers, I will change custom-response to support streaming response

Reply:
Using custom response to simulate responses will definitely add the content-length header. In real scenarios, when using the text/event-stream MIME type, it's not necessarily chunked transfer either; for example, if a proxy does caching in between, the content-length header might also be added. Therefore, the logic should still include compatibility handling and uniformly remove the content-length header.